### PR TITLE
Fix sata disk identifier SN

### PR DESCRIPTION
**Pull request by anhou**
when install ESXi,  lodash ‘_’  may be before or after disk SN,  like 'XXXXXX_____________' or '_____________YYYYYY' so the esxiWwid may be 
"esxiWwid":"t10.ATA_____QEMU_HARDDISK_XXXXXX_____________" or
"esxiWwid":"t10.ATA_____QEMU_HARDDISK______________YYYYYY"  it's decided by the raw data read by 'hdparm', previously, lodash ’_‘ is fixed before 'SN' in codes. 
this issue is found when installing ESXi6.0 in InfraSIM. with this fix, ESX6.0 could be installed with InfraSIM.

depends on: https://github.com/RackHD/on-imagebuilder/pull/45

@RackHD/corecommitters @RackHD/rackhd_dev

### CODE REVIEW

File: data/templates/get_driveid.js [hunk 1]
```diff
@@ -3,15 +3,45 @@
...
 var cmdVdInfo = 'ls -l /dev/disk/by-path';
 var cmdScsiId = 'lsscsi';
+var cmdSataRawInfo = 'sudo hdparm --Istdout';
 var options = {
     timeout: 10000 //10 seconds
 };
 
 /**
+ * Get SATA drive's SN string.
+ * @param {String} SATA drive's name, eg. 'sda'
```
Comment (WangWinson) [inline data/templates/get_driveid.js]: the param name need to match, here is `sataDrive` ?

File: data/templates/get_driveid.js [hunk 2]
```diff
@@ -3,15 +3,45 @@
...
+ * @return {String} SATA drive's SN string with 20 characters like 'QM00007_____________'
+ */
+function getSataSnStr(sataDrive) {
+    var output = execSync(cmdSataRawInfo + ' /dev/' + sataDrive);
+    var lines = output.toString().split('\n');
+    var snHexStr = lines.reduce(function (result,line) {
```
Comment (WangWinson) [inline data/templates/get_driveid.js]: maybe add comments about the example data will be helpful.

**Comment by yyscamper** [comment]
:+1:

File: data/templates/get_driveid.js [hunk 3]
```diff
@@ -3,15 +3,64 @@
...
+     *  snHexStr is from 21 to 40 bytes, total 20 bytes, shown as below
+     *
+     *  .....
+     *            514d 3030 3030 3520 2020 2020
+     *  2020 2020 2020 2020
+     *  .....
+     */
+    var lines = output.toString().split('\n');
+    var snHexStr = lines.reduce(function (result,line) {
+        if(line.length === 39) {
```
Comment (WangWinson) [inline data/templates/get_driveid.js]: I think the logic can work here, 
but what if some corner case make some line length equal 39 coincidentally. ? Maybe check the line format is hex will be more safe ?

**Comment by WangWinson** [comment]
still :+1:

File: data/templates/get_driveid.js (hunk 3 above)
Comment (anhou) [inline data/templates/get_driveid.js]: Great, add more accurate judgement

**Comment by pengz1** [comment]
+1

**Pull request merged**